src/label_map.cpp

```diff
@@ -74,34 +75,36 @@ void LabelMap::modify_lmap(int narg, char **arg)
 
   int ntypes;
   std::vector<std::string> *labels;
-  if (!strcmp(arg[0],"atom")) {
+  const std::string tlabel(arg[0]);
+  if (tlabel == "atom") {
     ntypes = natomtypes;
     labels = &typelabel;
-  } else if (!strcmp(arg[0],"bond")) {
+  } else if (tlabel == "bond") {
     ntypes = nbondtypes;
     labels = &btypelabel;
-  } else if (!strcmp(arg[0],"angle")) {
+  } else if (tlabel == "angle") {
     ntypes = nangletypes;
     labels = &atypelabel;
-  } else if (!strcmp(arg[0],"dihedral")) {
+  } else if (tlabel == "dihedral") {
     ntypes = ndihedraltypes;
     labels = &dtypelabel;
-  } else if (!strcmp(arg[0],"improper")) {
+  } else if (tlabel == "improper") {
     ntypes = nimpropertypes;
     labels = &itypelabel;
   } else error->all(FLERR,"Illegal labelmap command");
 
   int itype;
   int iarg = 1;
-  char *charlabel;
   while (iarg < narg) {
     itype = utils::inumeric(FLERR,arg[iarg++],false,lmp);
-    charlabel = arg[iarg++];
     if (itype > ntypes)
       error->all(FLERR,"Topology type exceeds system topology type");
-    if (isdigit(charlabel[0]))
-      error->all(FLERR,"Type labels cannot start with a number");
-    (*labels)[itype-1] = charlabel;
+    std::string slabel(arg[iarg++]);
+    auto words = Tokenizer(slabel, "[]").as_vector();
```

Comment:
please revert this commit, and I will merge these changes if you are finished with this PR. this functionality will be incorporated into the 'type-labels-infer_types' branch

Reply:
@jrgissing do you mean to revert only this commit? please go ahead and do as you think is better. I think you have access to my fork. As soon as you merge these changes I can update your implementation for better speed up in case of a large number of bonds.

Reply:
yes, specifically the part that make assumptions about the type label format. okay, that sounds useful

Reply:
@jrgissing I reverted these changes